src/main/java/net/sf/jabref/importer/UnlinkedFilesCrawler.java

```diff
@@ -60,12 +61,24 @@ public CheckableTreeNode searchDirectory(File directory, UnlinkedPDFFileFilter f
             return null;
         }
 
-        List<File> files = Arrays.asList(directory.listFiles(ff));
+        File[] filesArray = directory.listFiles(ff);
```

Comment:
There's a problem if `directory.listFiles(ff)` returns `null`. If we know that it won't happen then the previous code can indeed be simplified further (but as I recall it, iterating over `null` doesn't work as well as iterating over an empty list). I got a warning in Eclipse, so that is why I tried to solve it.